country-a-service/epps-service/src/test/java/dk/nsp/epps/service/mapping/DispensationMapperTest.java

```diff
@@ -42,89 +58,110 @@ StartEffectuationResponseType testStartEffectuationResponse(Document cda) throws
             .build();
     }
 
-    @Test
-    void startEffectuationRequestTest() throws MapperException {
+    @ParameterizedTest
+    @MethodSource("testDispensationCdas")
+    void startEffectuationRequestTest(String xmlFileName, String mappingErrorMessageExpected) throws MapperException {
         var sut = new DispensationMapper();
-        var cda = testDispensationCda();
+        var cda = testDispensationCda(xmlFileName);
         var req = sut.startEffectuationRequest("1111111118^^^&2.16.17.710.802.1000.990.1.500&ISO", cda);
 
         assertThat(req.getPrescription().size(), is(equalTo(1)));
         Assertions.assertTrue(req.getPrescription().getFirst().getIdentifier() > 0L);
         assertThat(req.getPersonIdentifier().getValue(), is(equalTo("1111111118")));
     }
 
-    @Test
-    void authorPersonTest() throws XPathExpressionException {
+    @ParameterizedTest
+    @MethodSource("testDispensationCdas")
+    void authorPersonTest(String xmlFileName, String mappingErrorMessageExpected) throws XPathExpressionException {
         var sut = new DispensationMapper();
-        var cda = testDispensationCda();
+        var cda = testDispensationCda(xmlFileName);
         var person = sut.authorPerson(cda);
 
         Assertions.assertNotNull(person.getPersonIdentifier());
         Assertions.assertFalse(person.getName().getGivenName().isBlank());
         Assertions.assertFalse(person.getName().getSurname().isBlank());
     }
 
-    @Test
-    void authorRoleTest() throws XPathExpressionException {
+    @ParameterizedTest
+    @MethodSource("testDispensationCdas")
+    void authorRoleTest(String xmlFileName, String mappingErrorMessageExpected) throws XPathExpressionException {
         var sut = new DispensationMapper();
-        var cda = testDispensationCda();
+        var cda = testDispensationCda(xmlFileName);
         var role = sut.authorRole(cda);
 
         Assertions.assertFalse(role.isBlank());
     }
 
-    @Test
-    void authorOrganizationTest() throws XPathExpressionException {
+    @ParameterizedTest
+    @MethodSource("testDispensationCdas")
+    void authorOrganizationTest(String xmlFileName, String mappingErrorMessageExpected) throws XPathExpressionException {
         var sut = new DispensationMapper();
-        var cda = testDispensationCda();
+        var cda = testDispensationCda(xmlFileName);
         var org = sut.authorOrganization(cda);
 
+        //TODO Revisit these, they will probably not always be blank. Do we need to validate this? As long as the
+        // mapper succeeds, the test is functional. If we want to test values mapped correctly, we should probably
+        // either not do the parameterized tests, or we should include the expected values in the parameters.
```

Comment:
Just because some of our tests are parameterised doesn't mean that others can't be tailored to the specific request.

Reply:
Yes, I might have been overzealous in making them all parameterized. We should create some ones with interesting and specific data to test on, because these right now does not give much value (even if I was probably originally the one to make the assertions THIS bland, because we kept changing the test input.